core/state/statedb.go

```diff
@@ -712,6 +717,8 @@ func (s *StateDB) Copy() *StateDB {
 		journal:              newJournal(),
 		hasher:               crypto.NewKeccakState(),
 	}
+	// Initialize new multi-transaction snapshot stack for the copied state
+	state.multiTxSnapshotStack = NewMultiTxSnapshotStack(state)
```

Comment:
> Shouldn't Copy also copy the multi tx stack?
This is risky because it depends on when Copy is called, which we have less control over, similar to this salient comment already in Copy:
```
	// In practice, we only ever copy state _between_ transactions/blocks, never
	// in the middle of a transaction. However, it doesn't cost us much to copy
	// empty lists, so we do it anyway to not blow up if we ever decide copy them
	// in the middle of a transaction.
```

> stack be empty/finalized at the time of calling Copy
This seems reasonable, I can add a panic for this assertion for now. 

> separate Finalize being called that cleans the multi tx snapshot

Not sure what you mean here - we already have `MultiTxSnapshotRevert` and `MultiTxSnapshotCommit`

Reply:
Added assertion for copy to check stack is empty

Reply:
Ended up just adding support for snapshot copying

It makes more sense for the API because the journal is copied as well 

It was more code changes but also made it easier to test when doing block simulations

I'll probably need to update the comment but the copying is working based on unit tests